src/NameFieldSet.tsx

```diff
@@ -59,6 +66,11 @@ export class NamesFieldSet extends React.Component<INameFieldSetProps, INameFiel
     this.setState({ names: [...this.state.names, ""] });
   }
 
+  private remove = (index: number) => (event: React.MouseEvent<HTMLElement>) => {
+    this.state.names.splice(index, 1)
+    this.setState({names: this.state.names})
```

Comment:
This might be problematic with react, because you are not creating a new this.states.names, but mutating it. React might not detect a change if this happens.

See this: https://medium.freecodecamp.org/handling-state-in-react-four-immutable-approaches-to-consider-d1f5c00249d5

What I would do: 
const names = [... this.state.names].splice(index,1);
this.setState({names});

Btw, if you find this confusing, we can use a library to manage the state called Immutable.js, which would provide a version of immutable objects (that you cant make these mistakes).